server/src/main/java/io/crate/analyze/where/EqualityExtractor.java

```diff
@@ -407,8 +394,10 @@ public Symbol visitReference(Reference ref, Context context) {
             return super.visitReference(ref, context);
         }
 
+        @Override
         public Symbol visitFunction(Function function, Context context) {
-
+            // Normalize the function so any casts on literals are evaluated
+            function = context.normalize.apply(function);
```

Comment:
Why do we normalize functions only but not simply the whole query instead?

Reply:
You're right, I moved the normalization here, due to some other issue I had, but it's no longer necessary, I moved it to normalize the whole query instead.